command/connect/redirecttraffic/redirect_traffic.go

```diff
@@ -150,9 +153,25 @@ func (c *cmd) generateConfigFromFlags() (iptables.Config, error) {
 			}
 		}
 
-		svc, _, err := c.client.Agent().Service(c.proxyID, nil)
-		if err != nil {
-			return iptables.Config{}, fmt.Errorf("failed to fetch proxy service from Consul Agent: %s", err)
+		var svc *api.AgentService
+		if c.nodeName == "" {
+			var err error
```

Comment:
This might not be needed?

Reply:
This is needed for the call below.

Reply:
🤔 interesting, i thght it would just use https://github.com/hashicorp/consul/blob/main/command/connect/redirecttraffic/redirect_traffic.go#L145, but maybe I am missing something.

Reply:
🤦‍♀️ you're totally right! will remove.